src/Swift.Runtime/src/Swift/Runtime/ProtocolWitnessTable.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+
+namespace Swift.Runtime;
+
+public readonly struct ProtocolConformanceDescriptor : IEquatable<ProtocolConformanceDescriptor>
+{
+    private readonly IntPtr _handle;
+
+    private ProtocolConformanceDescriptor(IntPtr handle)
+    {
+        _handle = handle;
+    }
+
+    public readonly static ProtocolConformanceDescriptor Zero = default;
+
+    public bool IsValid => _handle != IntPtr.Zero;
+
+    public bool Equals(ProtocolConformanceDescriptor other)
+    {
+        return _handle == other._handle;
+    }
+
+    internal static bool TryGetProtocolConformanceDescriptor<T, U>([NotNullWhen(true)] out ProtocolConformanceDescriptor? result)
+    {
+        var type = typeof(T);
+
+        if (typeof(ISwiftObject).IsAssignableFrom(type))
+        {
+            var helperType = typeof(ProtocolConformanceDescriptorHelper<,>).MakeGenericType(typeof(T), typeof(U));
+            result = (ProtocolConformanceDescriptor)helperType.GetMethod("GetProtocolConformanceDescriptor")!.Invoke(null, null)!;
+            return true;
+        }
+
+        result = null;
+        return false;
+    }
+
+    public static ProtocolConformanceDescriptor LoadProtocolConformanceDescriptor(string libraryPath, string symbolName)
```

Comment:
Naming nit. The callsite as is would look like:
```C#
var pcd = ProtocolConformanceDescriptor.LoadProcotolConformanceDescriptor(libraryPath, "mangled_name");
```

Typically we want to avoid repeating the name of the class in a factory method, as it's already there in the type name. So either `Load` or `LoadFromSymbol` or something like that.